andela/heimdal-ah-backend
# #162171039 Functionality for users to highlight and comment on certain sections of an article

#### What does this PR do?
- This PR adds functionality for users to highlight portions of an article for purposes of reference or interest.
- It also adds functionality for a highlighted section to be commented on. Users can see all the highlighted sections of the article as well as fetch comments belonging to any particular highlight.

#### Description of Task to be completed?
- Users should be able to highlight portions of an article and leave a comment.
- Other users should be able see one or more highlighted sections within such articles.
- When a highlighted section is clicked, it should fetch all the comments in that section if any.
- If sections of the article is updated, it should adjust the highlighted sections accordingly or remove the highlights entirely.

#### How should this be manually tested?
- Clone this Repository to your local machine
- Run ```npm install```
- Run ```npm run start:dev```
- Signup a new user via postman through POST 'localhost:4000/api/v1/auth/signup'
- Get the returned token
- Add the token to the headers (access-token) and use postman to create an article via POST 'localhost:4000/api/v1/article. Keep track of the returned articles id
- Select and copy out any portion of the article. Hit this endpoint on postman 'localhost:4000/api/v1/articles/:articleId/highlights' with a Post request containing the highlighted text as a body. Remember to add the access-token to the headers.
- Your first highlight has been successfully registered. You can fetch the article just created to see all highlighted portions. GET 'api/v1/articles/:articleId'

#### What are the relevant pivotal tracker stories?
[#162171039](https://www.pivotaltracker.com/story/show/162171039)

#### Any background context you want to add?
- This new feature has certain limitations. For example, when the author updates sections of the article without duplicating highlighted sections, everything works fine. But if the author chooses to duplicate a highlighted section, unexpected results may arise.

#### Screenshots
